lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1836,6 +1837,11 @@ signTransaction tl preferredLatestEra witCountCtx keyLookup mextraRewardAcc
 type MakeRewardAccountBuilder k =
     (k 'RootK XPrv, Passphrase "encryption") -> (XPrv, Passphrase "encryption")
 
+data ErrWriteTxEra
+    = ErrOldEraNotSupported Cardano.AnyCardanoEra
+    | ErrTxNotInNodeEra Write.AnyRecentEra
+    deriving (Show, Eq)
```

Comment:
Could be moved to the `Write.Tx.Balance` module with a function like:

```
data EraDependentPair f g era = EraDependentPair (f era) (g era)

chooseEraForBalancing
    :: InAnyRecentEra PParams
    -> InOneOrMoreRecentEra Tx
    -> InAnyRecentEra (PParams `EraDependentPair` Tx)
chooseEraForBalancing = undefined
```

and we'd want to use it if we were to provide FFI bindings for `balanceTransaction`.